frontend/lib/wallets/MetaMask.ts

```diff
@@ -76,12 +76,15 @@ export default class MetaMask extends AbstractWallet {
         if (networkType) {
             await setSigner(networkType, signer as any);
         }
-        window.$nuxt.$store.dispatch('network/setWalletChainId', window.ethereum.chainId);
+        await window.$nuxt.$store.dispatch('network/setWalletChainId', window.ethereum.chainId);
     }
 
     public accountsChangedHandler(addresses: Array<string>) {
         this.addresses = addresses;
-        window.$nuxt.$store.dispatch('wallet/setAddress', this.address);
+        window.$nuxt.$store.dispatch('wallet/setAddress', this.address).then(
+            () => {},
+            () => {}
+        );
```

Comment:
```suggestion
        window.$nuxt.$store.dispatch('wallet/setAddress', this.address).catch(() => {});
```

Since the rule only requires us to catch things, I believe that will be a little shorter modification